mesatee_services/fns/sgx_trusted_lib/src/fns.rs

```diff
@@ -37,18 +37,25 @@ fn invoke_worker(
     // New worker context
     let worker_context = running_task.get_worker_context();
     let payload = request.payload.clone();
-    worker.prepare_input(payload, file_list)?;
-    let result = worker.execute(worker_context);
-    match result {
-        Ok(output) => {
-            let _ = running_task.save_dynamic_output(&output);
-            running_task.finish()?;
-            let response = InvokeTaskResponse::new(&output);
-            Ok(response)
+    match worker.prepare_input(payload, file_list) {
+        Ok(_) => {
+            let result = worker.execute(worker_context);
+            match result {
+                Ok(output) => {
+                    let _ = running_task.save_dynamic_output(&output);
+                    running_task.finish()?;
+                    let response = InvokeTaskResponse::new(&output);
+                    Ok(response)
+                }
+                Err(err) => {
+                    let _ = running_task.finish();
+                    Err(err)
+                }
+            }
         }
-        Err(err) => {
+        Err(_) => {
             let _ = running_task.finish();
-            Err(err)
+            Err(Error::from(ErrorKind::InvalidInputError))
```

Comment:
Can we not nest the `prepare_input` and `execute` match?

```
if worker.prepare_input(payload, file_list).is_err() {
   // finish task
   // return ...
}
```

BTW, @litongxin1991, do we have "failed" status for a task?

Reply:
I agree. We can use early return to avoid nested code.
```
if let Err(err) = worker.prepare_input(payload, file_list) {
    let _ = running_task.finish();
    return Err(err);
}
let result = worker.execute(worker_context);
.... 
```
Besides, although in most cases the error is ```InvalidInputError```,  the error may be other one. So it's better to keep the error information.

@mssun We do have the "failed status". In the current design,  if the task succeeds, there is  ```dynamic output``. So when the task is finished, fns will check whether the dynamic output exists and update task status accordingly.  In the future design, we need to add a new api related to task failure.

Reply:
Since this PR is not being updated by its author, I recommend we work on this.